File: 0938-range-sum-of-bst/0938-range-sum-of-bst.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func rangeSumBST(root *TreeNode, low int, high int) int {
    var answer int

	var dfs func(*TreeNode)
	dfs = func(node *TreeNode) {
		if node == nil {
			return
		}
		if node.Val >= low && node.Val <= high {
			answer += node.Val
		}
		dfs(node.Left)
		dfs(node.Right)
	}
	dfs(root)
	return answer
}

File: 0938-range-sum-of-bst/0938-range-sum-of-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int rangeSumBST(TreeNode root, int low, int high) {
        int[] answer = {0};
        dfs(answer, root, low, high);
        return answer[0];
    }

    static public void dfs(int[] answer, TreeNode node, int low, int high) {
        if (node == null) {
            return;
        }
        if (node.val >= low && node.val <= high) {
            answer[0] += node.val;
        }
        dfs(answer, node.left, low, high);
        dfs(answer, node.right, low, high);
    }
}